src/main/java/org/eclipse/tracecompass/trace_event_logger/AsyncFileHandler.java

```diff
@@ -0,0 +1,346 @@
+    private BlockingQueue<List<LogRecord>> fQueue;
+    private Thread fWriterThread;
+    private int fMaxSize = 1024;
+    private int fQueueDepth = 10000;
+    private int fFlushRate = 1000;
+    private String fEncoding;
+    private Filter fFilter;
+    private ErrorManager fErrorManager;
+    private Formatter fFormatter;
+    private Level fLevel;
+
+    private List<LogRecord> fRecordBuffer = new ArrayList<>(fMaxSize);
+    private Timer fTimer = new Timer(false);
+    TimerTask fTask = new TimerTask() {
+        @Override
+        public void run() {
+            if (!fRecordBuffer.isEmpty()) {
+                flush();
+            }
+        }
+    };
+
+    private void configure() {
+        LogManager manager = LogManager.getLogManager();
+
+        String cname = getClass().getName();
+        String prop = manager.getProperty(cname + ".maxSize"); //$NON-NLS-1$
+        fMaxSize = 1024;
+        try {
+            fMaxSize = Integer.parseInt(prop.trim());
+        } catch (Exception ex) {
+            // we tried!
+        }
+        if (fMaxSize < 0) {
+            fMaxSize = 1024;
+        }
+        fQueueDepth = 10000;
+        prop = manager.getProperty(cname + ".queueDepth"); //$NON-NLS-1$
+        try {
+            fQueueDepth = Integer.parseInt(prop.trim());
+        } catch (Exception ex) {
+            // we tried!
+        }
+        if (fQueueDepth < 0) {
+            fQueueDepth = 10000;
+        }
+        fFlushRate = 1000;
+        prop = manager.getProperty(cname + ".formatter"); //$NON-NLS-1$
+        try {
+            fFormatter = (Formatter) ClassLoader.getSystemClassLoader().loadClass(prop).getDeclaredConstructor().newInstance();
+        } catch (Exception e) {
+            // we tried!
+        }
+        prop = manager.getProperty(cname + ".filter"); //$NON-NLS-1$
+        try {
+            fFilter = (Filter) ClassLoader.getSystemClassLoader().loadClass(prop).getDeclaredConstructor().newInstance();
+        } catch (Exception e) {
+            // we tried!
+        }
+
+        prop = manager.getProperty(cname + ".encoding"); //$NON-NLS-1$
+        try {
+            setEncoding(prop);
+        } catch (Exception e) {
+            // we tried!
+        }
+
+        prop = manager.getProperty(cname + ".flushRate"); //$NON-NLS-1$
+        try {
+            fFlushRate = Integer.parseInt(prop.trim());
+        } catch (Exception ex) {
+            // we tried!
+        }
+        if (fFlushRate < 0) {
+            fFlushRate = 1000;
+        }
+    }
+
+    /**
+     * Default Constructor
+     *
+     * @throws SecurityException
+     *             if a security manager exists and if the caller does not have
+     *             LoggingPermission("control").
+     * @throws IOException
+     *             if there are IO problems opening the files.
+     */
+    public AsyncFileHandler() throws SecurityException, IOException {
+        this(""); // let's hope it's configured in logging //$NON-NLS-1$
```

Comment:
This causes an exception in line 196 when using a properties file:
Can't load log handler "org.eclipse.tracecompass.trace_event_logger.AsyncFileHandler"
java.lang.IllegalArgumentException
java.lang.IllegalArgumentException
	at java.util.logging.FileHandler.<init>(FileHandler.java:304)
	at org.eclipse.tracecompass.trace_event_logger.AsyncFileHandler.<init>(AsyncFileHandler.java:196)
        at org.eclipse.tracecompass.trace_event_logger.AsyncFileHandler.<init>(AsyncFileHandler.java:179)

Reply:
Does it still?